### PR TITLE
refactor webview auth default region setting, move from LastLoginIdclnfo constructor to callee(webview browsers)

### PR DESCRIPTION
**Pull request by Will-ShaoHua**
As a new `ToolkitAuthManagerState` will be created every few mins by the IDE to sync, thus the same warning log entry will pop up accordingly and spam the log file

```
data class ToolkitAuthManagerState(
    // TODO: can't figure out how to make deserializer work with polymorphic types
    var ssoProfiles: List<ManagedSsoProfile> = emptyList(),
    var lastLoginIdcInfo: LastLoginIdcInfo = LastLoginIdcInfo()
)

data class LastLoginIdcInfo(
    var profileName: String = "",
    var startUrl: String = "",
    var region: String = AwsRegionProvider.getInstance().defaultRegion().id
)
```

image

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Checklist
- [ ] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [ ] A short description of the change has been added to the **[CHANGELOG](https://github.com/aws/aws-toolkit-jetbrains/blob/master/CONTRIBUTING.md#contributing-via-pull-requests)** if the change is customer-facing in the IDE.
- [ ] I have added metrics for my changes (if required)
 
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by Will-ShaoHua** [comment]
not clean (duplicate code) but there is one WIP webview refactor PR #4405

File: plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/QLoginWebview.kt [hunk 1]
```diff
@@ -221,7 +221,12 @@ class QWebviewBrowser(val project: Project, private val parentDisposable: Dispos
         }
 
         // previous login
-        val lastLoginIdcInfo = ToolkitAuthManager.getInstance().getLastLoginIdcInfo()
+        val lastLoginIdcInfo = ToolkitAuthManager.getInstance().getLastLoginIdcInfo().apply {
+            // set default option as us-east-1
+            if (this.region.isBlank()) {
+                this.region = AwsRegionProvider.getInstance().defaultRegion().id
```
Comment (rli) [inline plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/QLoginWebview.kt:227]: could you also modify `AwsRegionProvider#defaultRegion` to not print a stacktrace?

**Comment by Will-ShaoHua** [comment]
flaky test failing in windows 232 will be fixed in #4580

**Review by rli — APPROVED** [review]

**Pull request merged**